src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx

```diff
@@ -69,6 +69,193 @@ class WidgetCard extends React.Component<Props> {
     return false;
   }
 
+  renderToolbar() {
+    if (!this.props.isEditing) {
+      return null;
+    }
+
+    if (this.props.hideToolbar) {
+      return <ToolbarPanel />;
+    }
+
+    const {onEdit, onDelete, startWidgetDrag} = this.props;
+
+    return (
+      <ToolbarPanel>
+        <IconContainer data-component="icon-container">
+          <StyledIconGrabbable
+            color="gray500"
+            size="md"
+            onMouseDown={event => startWidgetDrag(event)}
+            onTouchStart={event => startWidgetDrag(event)}
+          />
+          <IconClick
+            data-test-id="widget-edit"
+            onClick={() => {
+              onEdit();
+            }}
+          >
+            <IconEdit color="gray500" size="md" />
+          </IconClick>
+          <IconClick
+            data-test-id="widget-delete"
+            onClick={() => {
+              onDelete();
+            }}
+          >
+            <IconDelete color="gray500" size="md" />
+          </IconClick>
+        </IconContainer>
+      </ToolbarPanel>
+    );
+  }
+
+  render() {
+    const {
+      widget,
+      isDragging,
+      api,
+      organization,
+      selection,
+      renderErrorMessage,
+      location,
+      router,
+    } = this.props;
+    return (
+      <ErrorBoundary
+        customComponent={<ErrorCard>{t('Error loading widget data')}</ErrorCard>}
+      >
+        <StyledPanel isDragging={isDragging}>
+          <WidgetQueries
+            api={api}
+            organization={organization}
+            widget={widget}
+            selection={selection}
+          >
+            {({tableResults, timeseriesResults, errorMessage, loading}) => {
+              return (
+                <React.Fragment>
+                  {typeof renderErrorMessage === 'function'
+                    ? renderErrorMessage(errorMessage)
+                    : null}
+                  <ChartContainer>
+                    <HeaderTitleLegend>{widget.title}</HeaderTitleLegend>
+                    <WidgetCardVisuals
+                      timeseriesResults={timeseriesResults}
+                      tableResults={tableResults}
+                      errorMessage={errorMessage}
+                      loading={loading}
+                      location={location}
+                      widget={{
+                        ...widget,
+                        title: '',
```

Comment:
If you're overwriting the title here, do you need the logic in shouldComponentUpdate that also overwrites the title? Seems like this would be redundant.

Reply:
This prop was leftover from a squash/rebase maneuver. It should remain `widget={widget}`, and the title override was moved to `shouldComponentUpdate`